Tests/LibWeb/Text/input/abortsignal-timeout.html

```diff
@@ -7,10 +7,16 @@
         const signal = AbortSignal.timeout(timeout_milliseconds);
         signal.onabort = (event) => {
             const abort_event_time = performance.now();
```

Comment:
Reading through #6362, I think it's important to consider that `performance.now()` uses a high resolution time source (not coarse) and coarsens it by 5 or 100 _microseconds_ depending on whether cross-origin isolation comes into play.

It then seems like `AbortSignal.timeout()` is the offender here. I'd expect a timeout signal to always take _at least_ the specified amount of time, but never less than that. Do other browser trigger the timeout earlier than 10ms?

Reply:
Firefox' 9ms is something I could understand given that `performance.now()` returns milliseconds, and unfortunate rounding could come into play together with the 'coarsened' time. For example, 995.5 -> 1005.4 (9.9ms) would round to 996 -> 1005 (9ms).

Reply:
Adjusting for unfortunate rounding (e.g. time + 1) should be fine I guess? But seeing 8ms or 7ms would really surprise me and hints at an issue with the way we trigger that timeout imo.

Reply:
FYI, looks like `WindowOrWorkerGlobalScopeMixin::run_steps_after_a_timeout_impl()` is barely implemented to spec. It looks like it should wait at least the given amount of time, which we clearly do not do at this point. That's probably our current timer's implementation fault?

At any rate, the test seems correct, we seem to be missing a correct implementation.